modules/connectors/hay_post/karrio/providers/hay_post/shipment/create.py

```diff
@@ -0,0 +1,119 @@
+        shipment = _extract_details(response, settings)
+    else:
+        messages = error.parse_error_response(response, settings)
+
+    return shipment, messages
+
+
+def _extract_details(
+    data: dict,
+    settings: provider_utils.Settings,
+) -> models.ShipmentDetails:
+    shipment = lib.to_object(hay_post_response.OrderCreateResponseType, data)
+    return models.ShipmentDetails(
+        carrier_id=settings.carrier_id,
+        carrier_name=settings.carrier_name,
+        tracking_number=shipment.barcode,
+        shipment_identifier=str(shipment.id),
+        docs=models.Documents(
+            label="No label...",
+        ),
+        meta=dict(
+            revertOrderId=shipment.revertOrderId,
+            revertBarcode=shipment.revertBarcode,
+            postalcode=shipment.postalcode,
+        ),
+    )
+
+
+def shipment_request(
+    payload: models.ShipmentRequest,
+    settings: provider_utils.Settings,
+) -> lib.Serializable:
+    shipper = lib.to_address(payload.shipper)
+    recipient = lib.to_address(payload.recipient)
+    packages = lib.to_packages(payload.parcels)
+    service = provider_units.ShippingService.map(payload.service)
+
+    options = lib.to_shipping_options(
+        payload.options,
+        package_options=packages.options,
+        initializer=provider_units.shipping_options_initializer,
+    )
+
+    additional_services = [
+        int(service.value.code) for service in provider_units.ShippingOption
+        if service.name in options
+    ]
+
+    shipper_country_id = next(
+        (
+            int(country.value)
+            for country in list(provider_units.ShippingCountries)
+            if shipper.country_code == country.name
+        ),
+        int(provider_units.ShippingCurrency.AMD.value),
+    )
+
+    recipient_country_id = next(
+        (
+            int(country.value)
+            for country in list(provider_units.ShippingCountries)
+            if recipient.country_code == country.name
+        ),
+        int(provider_units.ShippingCurrency.AMD.value),
+    )
+
+    request = hay_post.OrderCreateRequestType(
+        customerId=int(settings.customer_id),
+        serviceCategoryDirectionId=service.value,
+        weight=packages.weight.value,
+        # returnRegisteredAddress=True,
+        destinationAddress=hay_post.NAddressType(
+            countryId=recipient_country_id,
+            street=f"{recipient.street} {recipient.address_line1}",
+            cityVillage=recipient.city,
+            postalCode=recipient.postal_code,
+            receiverInfo=hay_post.ReceiverInfoType(
+                companyName=recipient.company_name,
+                firstName=recipient.person_name,
+                phoneNumber=recipient.phone_number,
+                email=recipient.email,
+            ),
+        ),
+        returnAddress=hay_post.NAddressType(
+            countryId=shipper_country_id,
+            street=f"{shipper.street} {shipper.address_line1}",
```

Comment:
```suggestion
            street=shipper.street,
```

same comment as above